c/ntdrivers/parport.i.cil-2.c

```diff
@@ -15,7 +15,7 @@ extern unsigned long __VERIFIER_nondet_ulong(void);
 extern void abort (void) __attribute__ ((__nothrow__ , __leaf__)) __attribute__ ((__noreturn__));
 void memcpy_guard(void* p1, void* p2, unsigned int n)
 {
-    if ((char*)p1 + n <= (char*)p2 || (char*)p2 + n <= (char*)p1)
+    if ((unsigned long)p1 + n <= (unsigned long)p2 || (unsigned long)p2 + n <= (unsigned long)p1)
```

Comment:
Why was this needed? It does not look like anything related to initialization.

Reply:
Ah, yes, I need to break this up. I first thought only a small number of fixes was necessary... The above change is to address undefined behaviour in comparing pointers over distinct objects. The cast to an integer-typed variable turns this into implementation-defined behaviour.